front/components/assistant_builder/AssistantBuilderDataSourceModal.tsx

```diff
@@ -138,19 +139,63 @@ function PickDataSource({
   show: boolean;
   onPick: (dataSource: DataSourceType) => void;
 }) {
+  const [query, setQuery] = useState("");
+
+  // Order in the following format : connectorProvider > empty > webcrawler
+  const orderedDataSources = dataSources.sort((a, b) => {
```

Comment:
Can you move this function in `lib/assistant.ts` where we already have some assistant / related sorting. Name LGTM :+1: